homeassistant/components/fronius/sensor.py

```diff
@@ -159,11 +159,11 @@ async def fetch_data(*_):
 class FroniusAdapter:
     """The Fronius sensor fetching component."""
 
-    def __init__(self, bridge: Fronius, name, device, add_entities):
+    def __init__(self, bridge, name, device, add_entities):
         """Initialize the sensor."""
-        self.bridge = bridge
-        self._name = name
-        self._device = device
+        self.bridge: Fronius = bridge
+        self._name: str = name
+        self._device: int = device
```

Comment:
Why did you add typing to these? You could just type the method arguments instead?

Reply:
No because I dont want to type "add_entities", which I would have to type if I type any argument...

Reply:
So, uh... why not type `add_entities`? 😕

Reply:
Because it is the passed on function async_add_entities, that originates from somewhere within the home assistant core. I don't know its type, the type is not really relevant IMO and is probably so huge that it makes the whole method head unreadable :/

Reply:
Well, this does not improve it either. Let me suggest the required types in another comment

Reply:
Meh, GitHub doesn't allow me to provide it as suggestion, however this is the code:

```py
    def __init__(
        self, bridge: Fronius, name: str, device: int, add_entities: AddEntitiesCallback
    ) -> None:
        """Initialize the sensor."""
        self.bridge = bridge
        self._name = name
        self._device = device
```

Also add this import in the top of the file:

```py
from homeassistant.helpers.entity_platform import AddEntitiesCallback
```

Reply:
Great, thanks! :)

Reply:
You really don't need to type these anymore, as they are typed from the method already

```suggestion
        self.bridge = bridge
        self._name = name
        self._device = device
```